mathext/ell_carlson.go

```diff
@@ -0,0 +1,145 @@
+		tol   = 1.2674918778210762260320167734407048051023273568443e-02 // (3ε)^(1/8)
+	)
+	if x < 0 || y < 0 || z < 0 || math.IsNaN(x) || math.IsNaN(y) || math.IsNaN(z) {
+		return math.NaN()
+	}
+	if upper < x || upper < y || upper < z {
+		return math.NaN()
+	}
+	if x+y < lower || y+z < lower || z+x < lower {
+		return math.NaN()
+	}
+
+	A0 := (x + y + z) / 3
+	An := A0
+	Q := math.Max(math.Max(math.Abs(A0-x), math.Abs(A0-y)), math.Abs(A0-z)) / tol
+	xn, yn, zn := x, y, z
+	mul := 1.0
+
+	for Q >= mul*math.Abs(An) {
+		xnsqrt, ynsqrt, znsqrt := math.Sqrt(xn), math.Sqrt(yn), math.Sqrt(zn)
+		lambda := xnsqrt*ynsqrt + ynsqrt*znsqrt + znsqrt*xnsqrt
+		An = (An + lambda) * 0.25
+		xn = (xn + lambda) * 0.25
+		yn = (yn + lambda) * 0.25
+		zn = (zn + lambda) * 0.25
+		mul *= 4
+	}
+
+	X := (A0 - x) / (mul * An)
+	Y := (A0 - y) / (mul * An)
+	Z := -(X + Y)
+	E2 := X*Y - Z*Z
+	E3 := X * Y * Z
+
+	// http://dlmf.nist.gov/19.36.E1
+	return (E3*(6930*E3+E2*(15015*E2-16380)+17160) + E2*((10010-5775*E2)*E2-24024) + 240240) / (240240 * math.Sqrt(An))
+}
+
+// EllipticRD computes the symmetric elliptic integral R_D(x,y,z):
+//
+//	R_D(x,y,z) = (1/2)\int_{0}^{\infty}{1/(s(t)(t+z))} dt,
+//	s(t) = \sqrt{(t+x)(t+y)(t+z)}.
+//
+// See: http://dlmf.nist.gov/19.16.E5 for the definition.
+//
+// See: http://doi.org/10.1145/355958.355970 for the original Fortran code.
+//
+// See: http://dx.doi.org/10.1007/BF02198293 for the modified method of computation.
+func EllipticRD(x, y, z float64) float64 {
+	const (
+		lower = 4.8095540743116787026618007863123676393525016818363e-103 // (5*2^-1022)^(1/3)
+		upper = 1 / lower
+		tol   = 9.0351169339315770474760122547068324993857488849382e-03 // (ε/5)^(1/8)
+	)
+	if x < 0 || y < 0 || math.IsNaN(x) || math.IsNaN(y) || math.IsNaN(z) {
+		return math.NaN()
+	}
+	if upper < x || upper < y || upper < z {
+		return math.NaN()
+	}
+	if x+y < lower || z < lower {
+		return math.NaN()
+	}
+
+	A0 := (x + y + 3*z) / 5
+	An := A0
+	Q := math.Max(math.Max(math.Abs(A0-x), math.Abs(A0-y)), math.Abs(A0-z)) / tol
+	xn, yn, zn := x, y, z
+	mul, s := 1.0, 0.0
+
+	for Q >= mul*math.Abs(An) {
+		xnsqrt, ynsqrt, znsqrt := math.Sqrt(xn), math.Sqrt(yn), math.Sqrt(zn)
+		lambda := xnsqrt*ynsqrt + ynsqrt*znsqrt + znsqrt*xnsqrt
+		s += 1 / (mul * znsqrt * (zn + lambda))
+		An = (An + lambda) * 0.25
+		xn = (xn + lambda) * 0.25
+		yn = (yn + lambda) * 0.25
+		zn = (zn + lambda) * 0.25
+		mul *= 4
+	}
+
+	X := (A0 - x) / (mul * An)
+	Y := (A0 - y) / (mul * An)
+	Z := -(X + Y) / 3
+	E2 := X*Y - 6*Z*Z
+	E3 := (3*X*Y - 8*Z*Z) * Z
+	E4 := 3 * (X*Y - Z*Z) * Z * Z
+	E5 := X * Y * Z * Z * Z
+
+	// http://dlmf.nist.gov/19.36.E2
+	return ((471240-540540*E2)*E5+(612612*E2-540540*E3-556920)*E4+E3*(306306*E3+E2*(675675*E2-706860)+680680)+E2*((417690-255255*E2)*E2-875160)+4084080)/(4084080*mul*An*math.Sqrt(An)) + 3*s
+}
+
+// EllipticF computes the Legendre's elliptic integral of the 1st kind F(\phi|m):
+//
+//	F(\phi|m) = \int_{0}^{\phi}1 / \sqrt{1-m\sin^2\theta} d\theta
```

Comment:
Could you please put a space before `1 /` and put the argument of sine into parenthesis? Also below.